tools/src/main/java/org/apache/kafka/tools/ConsumerPerformance.java

```diff
@@ -134,8 +136,13 @@ private static void consume(KafkaConsumer<byte[], byte[]> consumer,
         long reportingIntervalMs = options.reportingIntervalMs();
         boolean showDetailedStats = options.showDetailedStats();
         SimpleDateFormat dateFormat = options.dateFormat();
-        consumer.subscribe(options.topic(),
-            new ConsumerPerfRebListener(joinTimeMs, joinStartMs, joinTimeMsInSingleRound));
+
+        ConsumerPerfRebListener listener = new ConsumerPerfRebListener(joinTimeMs, joinStartMs, joinTimeMsInSingleRound);
```

Comment:
Perhaps `ConsumerPerfRebListener` needs to be updated. It expects the fetch can only work if partitions are assigned, so it calculate the elapsed time of the join in `onPartitionsAssigned`. However, in the `include` mode, other topics could be assigned to this consumer even though the fetch is starting. That could ultimately corrupt the `fetchTimeInMs`

Reply:
I see what you mean, but isn't that the same dynamic partition assignment mechanism that we have when subscribing using a list of topics?

My understanding is that `fetchTimeInMs` can only be trusted when subscribed partitions remains stable during the entire test run, or when you are using classic rebalance protocol with a non cooperative assignment strategy. This makes sense, because this tool was created when stop-the-world rebalancing was the only option.